File: frontend/src/services/groupService.js
import api from './api';

export const groupService = {
  getAllGroups: async () => {
    const response = await api.get('/groups');
    return response.data;
  },

  createGroup: async (groupData) => {
    const response = await api.post('/groups', groupData);
    return response.data;
  },

  getGroupsForTeacher: async (teacherId) => {
    const response = await api.get(`/groups/teacher/${teacherId}`);
    return response.data;
  },

  addTeacherToGroup: async (groupId, teacherId) => {
    await api.post(`/groups/${groupId}/teachers/${teacherId}`);
  },

  removeTeacherFromGroup: async (groupId, teacherId) => {
    await api.delete(`/groups/${groupId}/teachers/${teacherId}`);
  },
};

File: src/main/java/com/example/edusence_studio/services/groups/GroupService.java
package com.example.edusence_studio.services.groups;

import com.example.edusence_studio.dtos.CreateGroupRequest;
import com.example.edusence_studio.models.groups.Group;

import java.util.List;
import java.util.UUID;

public interface GroupService {

    Group createGroup(CreateGroupRequest request);

    void addTeacherToGroup(UUID groupId, UUID teacherId);

    void removeTeacherFromGroup(UUID groupId, UUID teacherId);

    List<Group> getAllGroups();

    List<Group> getGroupsForTeacher(UUID teacherId);
}

